### DISCUSSION TITLE
Auto dark mode for all web sites.

### DISCUSSION THREAD

**Discussion by ssleert** [minbrowser/min#2386] [Ideas]
Hi guys, I recently switched back to a **dark color theme**, and realized that `Min` doesn't know how to automatically *convert* **white** sites to the **dark theme**. 

But as far as I *remember* `chromium` has an automatic **dark mode** for any site, it can even be enabled in `dev tools`.

How *hard* would it be to *integrate* this into `Min`? Should I try it?

<hr>

## Example of site without dark theme.
![image](https://github.com/minbrowser/min/assets/68077937/94c8c2e4-0c62-443f-86ed-142c8e992150)

<br>

## And with *auto* **dark theme** through `dev tools`.
![image](https://github.com/minbrowser/min/assets/68077937/9ed687bd-0b82-4c4c-877b-4c03fecb17bd)

<hr>

**Comment by ssleert** [comment]
@PalmerAL thank you in advance for your reply :)

**Comment by PalmerAL** [comment]
Seems like a reasonable thing to support (we have a userscript for this: https://github.com/PalmerAL/min-dark-mode-userscript, but the Chromium version probably works better).

Either Electron needs to expose a setting for this, or there needs to be a command-line argument for this in Chromium that we can turn on Do you know if there is one?

**Reply by ssleert** [reply]
Yes, you can set auto dark mode via cli flag
```
flatpak run org.chromium.Chromium --enable-features=WebContentsForceDark 
```
or with background lightness threshhold (the best mode as i think)
```
flatpak run org.chromium.Chromium --enable-features=WebContentsForceDark:background_lightness_threshold 
```
![Screenshot from 2024-02-06 12-06-20](https://github.com/minbrowser/min/assets/68077937/bf077feb-1194-4ee0-bbe9-685b554c19d4)

**Reply by PalmerAL** [reply]
Cool, in that case you could try getting it to work in Electron. We create new views here: https://github.com/minbrowser/min/blob/master/main/viewManager.js#L11. My guess is that this is settable through either "enableBlinkFeatures" or "additionalArguments": https://www.electronjs.org/docs/latest/api/browser-window

**Reply by 097115** [reply]
@PalmerAL, thanks for the hints!

Just to clarify: you mean adding something like this `additionalArguments: ['--enable-features=WebContentsForceDark'],` to the return dict of getDefaultViewWebPreferences()?

**Reply by 097115** [reply]
@ssleert, have you succeeded may be in adding this to Min?
